### PR TITLE
RUMM-1455 Add Session Start Callback

### LINKED ISSUE #526: [Feature Request] Get session ID
### [Feature Request] Get session ID 

Hi guys, I would like to know if there is a easy way to get the session ID. 

I wanted it to become available so we can associate it with other trackers, or became available in screen when debugging/testing the app.

Currently I'm doing something like this
```
Datadog.Configuration.builderUsing(...)
.setRUMViewEventMapper { viewEvent in
  let currentSessionID = viewEvent.session.id
  guard sessionID != currentSessionID else { return viewEvent }
  sessionID = currentSessionID
  return viewEvent
}
.build()
```

Thanks in advance!

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Adds an optional `onSessionStart` callback to `RUMApplicationScope` to be called when a new session starts. The callback takes 2 arguments: the newly started Session ID and a boolean indicating whether or not the session is discarded by the sampling rate (when `true` it means no event in this session will be kept).

Fix #526

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

**Comment by maxep** [comment]
This PR also surface API from recent PRs.

File: Sources/Datadog/DatadogConfiguration.swift [hunk 1]
```diff
@@ -505,6 +507,17 @@ extension Datadog {
                 return self
             }
 
+            /// Sets the Session start callback.
+            ///
+            /// The callback takes 2 arguments: the newly started Session ID and a boolean indicating whether or not the session is discarded by the sampling rate
+            /// (when `true` it means no event in this session will be kept).
+            ///
+            /// - Parameter handler: the callback handler to notify whenever a new Session starts.
+            public func onSessionStart(_ handler: @escaping (String, Bool) -> Void) -> Builder {
```
Comment (ncreated) [inline Sources/Datadog/DatadogConfiguration.swift:516]: - [ ] About the naming - as this is a RUM feature, I think it's worth expressing it from the name as we do for other RUM options. How about `onRUMSessionStart`?
- [ ] About testing - it also requires updating tests in [`DatadogConfigurationBuilderTests`](https://github.com/DataDog/dd-sdk-ios/blob/f571f041edb2b1097b6c4fd28bdcdfdf2408ca66/Tests/DatadogTests/Datadog/DatadogConfigurationBuilderTests.swift#L15) (we cover all public options in there)

**Review by ncreated — CHANGES_REQUESTED** [review]

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMApplicationScope.swift [hunk 2]
```diff
@@ -87,6 +92,7 @@ internal class RUMApplicationScope: RUMScope, RUMContextProvider {
     private func refresh(expiredSession: RUMSessionScope, on command: RUMCommand) {
         let refreshedSession = RUMSessionScope(from: expiredSession, startTime: command.time)
         sessionScope = refreshedSession
+        sessionScopeDidUpdate(refreshedSession)
```
[the remark this answers is not in the document]
Comment (maxep) [reply Sources/Datadog/RUM/RUMMonitor/Scopes/RUMApplicationScope.swift:95]: Yes, I agree 👍

**Comment by michaelhandkins** [comment]
Hey guys ✌️ I saw that this PR seemed to be in response to [this post](https://github.com/DataDog/dd-sdk-ios/issues/526) about accessing the session ID in code, but I can't figure out how to set that `onSessionStart` callback. All avenues I've tried say they're internally protected or can't be found.

**Pull request merged**